### DISCUSSION TITLE
Send a file to a http address

### DISCUSSION THREAD

**Discussion by Jooeeyy** [nextgenhealthcare/connect#5174] [Q&A]
Hey,
I'm still relatively new to Mirth and currently trying to cover the following scenario.

Reading CSV file with File Reader and then sending it to my website with HTTP Sender. Thereby the CSV file should be passed as FormFile.

Accordingly, I have a few query parameters in the form and for this the form should arrive in C# as Request.Form.Files. Unfortunately, I have the problem that I can not manage that it really transmits the CSV file and not the content. How to get them done?

![image](https://user-images.githubusercontent.com/7760232/166871368-40c23cb1-ba25-44e9-8c72-0d0d7414ab1e.png)
![image](https://user-images.githubusercontent.com/7760232/166871431-aa792f8d-896c-4820-a8bc-ffbc620b1d9d.png)
![image](https://user-images.githubusercontent.com/7760232/166871500-e6e7097f-ae69-44a1-9c52-abebfce59ad8.png)

I have already seen that in other blog entries a boundary still had to be attached. There it was done with a Destination Transformer. Unfortunately I don't know how to get the file from the file reader.`

-------------------------------------------------------------

var f = // How do i get the file from fileReader??

var boundary = '----------' + UUIDGenerator.getUUID();
$co('boundary', boundary); // make it available in destination as velocity template variable ${boundary} to be used in field "Content Type"

// http://www.mirthproject.org/community/forums/showthread.php?t=219080
// https://hc.apache.org/httpcomponents-client-4.5.x/httpmime/apidocs/org/apache/http/entity/mime/MultipartEntityBuilder.html#addBinaryBody(java.lang.String,%20java.io.File,%20org.apache.http.entity.ContentType,%20java.lang.String)

var name = 'file';
var contentType = org.apache.http.entity.ContentType.create('image/jpeg');
var fileName = 'IMG001.JPG';

// create multipart with the following data:
// Content-Disposition: form-data; name="file"; filename="IMG001.JPG"
// Content-Type: image/jpeg
var entity = org.apache.http.entity.mime.MultipartEntityBuilder.create()
	.setBoundary(boundary)
	.addBinaryBody(name, f, contentType, fileName)
	.build();
    
var bos = new java.io.ByteArrayOutputStream();
entity.writeTo(bos);
msg = FileUtil.encode(bos.toByteArray());

-------------------------------------------------------

Thank you very much!!

**Comment by jonbartels** [comment]
It looks like you got the code from https://forums.mirthproject.io/forum/mirth-connect/support/18273-multipart-form-post-of-base64-encoded-pdf-solved#post104394 . GOOD! Thats a good example.

I think if you look in your maps you will see map values for `originalFileName` and maybe a path. You can construct the `var fileName` from the file path in your CSV reader plus `originalFileName`. This will read the CSV from the same file as the file reader and send it.

If you are mapping or using MC to make changes to the CSV to a different format then you have to do something different. Look at https://hc.apache.org/httpcomponents-client-4.5.x/current/httpmime/apidocs/org/apache/http/entity/mime/MultipartEntityBuilder.html . You can do a few things:
* addTextBody *might* work for CSV or XML or other textual formats
* addBinaryBody has a variant that takes an input stream. You can get your content as a string (base64 encoded as needed) then convert it to an InputStream
